commit df4d69feaf017cd2d6d521d5d32870e7e05f7121
Author: Christophe Leroy <christophe.leroy@c-s.fr>
Date:   Tue Nov 21 15:28:20 2017 +0100

    powerpc: Fix boot on BOOK3S_32 with CONFIG_STRICT_KERNEL_RWX
    
    commit 252eb55816a6f69ef9464cad303cdb3326cdc61d upstream.
    
    On powerpc32, patch_instruction() is called by apply_feature_fixups()
    which is called from early_init()
    
    There is the following note in front of early_init():
     * Note that the kernel may be running at an address which is different
     * from the address that it was linked at, so we must use RELOC/PTRRELOC
     * to access static data (including strings).  -- paulus
    
    Therefore, slab_is_available() cannot be called yet, and
    text_poke_area must be addressed with PTRRELOC()
    
    Fixes: 95902e6c8864 ("powerpc/mm: Implement STRICT_KERNEL_RWX on PPC32")
    Reported-by: Meelis Roos <mroos@linux.ee>
    Signed-off-by: Christophe Leroy <christophe.leroy@c-s.fr>
    Signed-off-by: Michael Ellerman <mpe@ellerman.id.au>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/arch/powerpc/lib/code-patching.c b/arch/powerpc/lib/code-patching.c
index c9de03e0c1f1..d469224c4ada 100644
--- a/arch/powerpc/lib/code-patching.c
+++ b/arch/powerpc/lib/code-patching.c
@@ -21,6 +21,7 @@
 #include <asm/tlbflush.h>
 #include <asm/page.h>
 #include <asm/code-patching.h>
+#include <asm/setup.h>
 
 static int __patch_instruction(unsigned int *addr, unsigned int instr)
 {
@@ -146,11 +147,8 @@ int patch_instruction(unsigned int *addr, unsigned int instr)
 	 * During early early boot patch_instruction is called
 	 * when text_poke_area is not ready, but we still need
 	 * to allow patching. We just do the plain old patching
-	 * We use slab_is_available and per cpu read * via this_cpu_read
-	 * of text_poke_area. Per-CPU areas might not be up early
-	 * this can create problems with just using this_cpu_read()
 	 */
-	if (!slab_is_available() || !this_cpu_read(text_poke_area))
+	if (!this_cpu_read(*PTRRELOC(&text_poke_area)))
 		return __patch_instruction(addr, instr);
 
 	local_irq_save(flags);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1161_df4d69feaf017cd2d6d521d5d32870e7e05f7121.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1161_df4d69feaf017cd2d6d521d5d32870e7e05f7121.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
